sled-agent/src/services.rs

```diff
@@ -551,7 +597,33 @@ enum SledLocalZone {
     },
 }
 
-type ZoneMap = BTreeMap<String, RunningZone>;
+// The return type for `start_omicron_zones`.
+//
+// When multiple zones are started concurrently, some can fail while others
+// succeed. This structure allows the function to return this nuanced
+// information.
+#[must_use]
+struct StartZonesResult {
+    // The set of zones which have successfully started.
+    zones: Vec<OmicronZone>,
```

Comment:
Tiny nit - maybe name this field indicating success (`zones_started` or `successful_zones` or somesuch)? I could imagine thinking that this is "the whole set of zones we tried to start" if I were skimming through this.

Reply:
Renamed to `new_zones`!